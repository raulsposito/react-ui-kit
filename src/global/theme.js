// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
