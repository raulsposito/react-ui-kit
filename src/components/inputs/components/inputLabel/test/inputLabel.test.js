import React from 'react'
import { render, screen } from '@testing-library/react'

import { colors } from '@global/theme'

import InputLabel from '..'

const mockData = {
  text: 'Input Label',
  customColor: 'secondary',
  name: 'labelTest',
  requiredText: 'Input Label *'
}

describe('<InputLabel />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<InputLabel text={mockData.text} color={mockData.customColor} />)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('Text', () => {
  it('has text', () => {
    render(<InputLabel text={mockData.text} />)

    const text = screen.queryByText(mockData.text)
    expect(text).toBeInTheDocument()
  })
})

describe('Required', () => {
  it('is required', () => {
    render(<InputLabel text={mockData.text} required />)

    const requiredLabel = screen.queryByText(mockData.requiredText)
    expect(requiredLabel).toBeInTheDocument()
  })
})

describe('Name', () => {
  it('has name', () => {
    render(<InputLabel text={mockData.text} name={mockData.name} />)

    const label = screen.queryByText(mockData.text).closest('label')
    expect(label).toBeInTheDocument()
    expect(label).toHaveAttribute('for', mockData.name)
  })
})

describe('Color', () => {
  it('has color', () => {
    render(<InputLabel text={mockData.text} color={mockData.customColor} />)

    const label = screen.queryByText(mockData.text)
    expect(label).toHaveStyle({ color: `${colors[mockData.customColor].regular}` })
  })

  it('has not color', () => {
    render(<InputLabel text={mockData.text} />)

    const label = screen.queryByText(mockData.text)
    expect(label).toHaveStyle({ color: `${colors.contrast.regular}` })
  })
})
