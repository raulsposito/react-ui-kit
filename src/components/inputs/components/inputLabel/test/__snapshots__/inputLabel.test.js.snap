// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputLabel /> should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="sc-iBPTik gAnlej"
  >
    <p
      class="sc-bdfBQB AqsHc"
      color="secondary"
    >
      Input Label 
    </p>
  </label>
</DocumentFragment>
`;
